src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/events/InboundEvents.kt

```diff
@@ -0,0 +1,68 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service.events
+
+import com.fasterxml.jackson.annotation.JsonGetter
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import java.time.Instant
+
+enum class InboundEventType(val eventType: String) {
+  CIAG_INDUCTION_CREATED("ciag-induction.created") {
+    override fun toInboundEvent(mapper: ObjectMapper, message: String) =
+      mapper.readValue<CiagInductionCreatedEvent>(message)
+  },
+  CIAG_INDUCTION_UPDATED("ciag-induction.updated") {
+    override fun toInboundEvent(mapper: ObjectMapper, message: String) =
+      mapper.readValue<CiagInductionUpdatedEvent>(message)
+  },
+  ;
+
+  abstract fun toInboundEvent(mapper: ObjectMapper, message: String): InboundEvent
+}
+
+interface InboundEvent {
+  fun eventType(): String
+  fun reference(): String
+  fun prisonNumber(): String
+  fun occurredAt(): Instant
+}
+
+// ------------ CIAG Induction events ------------------------------------------
+abstract class CiagInductionEvent(
+  private val personReference: PersonReference,
+  private val additionalInformation: InductionInformation,
+) : InboundEvent {
+  override fun reference() = additionalInformation.reference
+  override fun prisonNumber(): String = personReference.identifiers.first { it.type == "NOMS" }.value
+  override fun occurredAt() = additionalInformation.occurredAt
+  fun prisonId() = additionalInformation.prisonId
+  fun userId() = additionalInformation.userId
+  fun userDisplayName() = additionalInformation.userDisplayName
+}
+
+data class CiagInductionCreatedEvent(
```

Comment:
What are you thoughts on splitting some of these classes/interfaces/enums out into their own source files? I'm really struggling to read and understand this source file I'm afraid 😢

Reply:
I'll take another look. I've been so focused on getting this stuff to work, that I haven't had my eyes on the actual code, which I've basically copied & pasted from activities-management.